### PR TITLE
PLATFORM-7238 restore our custom changes

### PR DESCRIPTION
**Pull request by mmeller-wikia**
https://fandom.atlassian.net/browse/PLATFORM-7238

Compare to https://github.com/Wikia/thumbor/compare/8b550f55ed10ebacadb8bf6ed624c3c400672746...master
See as base https://github.com/Wikia/smart-crop/pull/55

### CODE REVIEW

File: thumbor/detectors/animeface_detector/__init__.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# Fandom-change-start: add custom animeface_detector
+from typing import Dict
+
+from thumbor.detectors.local_detector import CascadeLoaderDetector
+
+HAIR_OFFSET = 0.12
+
+
+class Detector(CascadeLoaderDetector):
```
Comment (mmeller-wikia) [inline thumbor/detectors/animeface_detector/__init__.py:19]: I didn't find any logs logged out from our custom change in last 7 days. Let's simplify this filter. See  https://console.cloud.google.com/logs/query;query=%22Error%20during%20face%20detection%22;duration=P7D?referrer=search&project=fandom-prod
Original code:
```
class Detector(CascadeLoaderDetector):

    def __init__(self, context, index, detectors):
        super(Detector, self).__init__(context, index, detectors)
        self.load_cascade_file(__file__, self.context.config.ANIME_FACE_DETECTOR_CASCADE_FILE)

    def __add_hair_offset(self, top, height):
        top = max(0, top - height * HAIR_OFFSET)
        return top

    def detect(self, callback):
        try:
            features = self.get_features()
        except Exception as e:
            logger.exception(e)
            logger.warn('Error during face detection; skipping to next detector')
            self.next(callback)
            return

        if features:
            for (left, top, width, height), neighbors in features:
                top = self.__add_hair_offset(top, height)
                self.context.request.focal_points.append(
                    FocalPoint.from_square(left, top, width, height, origin="Anime Face Detection")
                )
            callback()
        else:
            self.next(callback)
```

File: thumbor/engines/pil.py [hunk 2]
```diff
@@ -525,12 +527,19 @@ def image_data_as_rgb(self, update_image=True):
 
         return converted_image.mode, converted_image.tobytes()
 
-    def convert_to_grayscale(self, update_image=True, alpha=True):
+    # Fandom-change-start: Make image smooth before running feace detection
+    def convert_to_grayscale(self, update_image=True, alpha=True, with_smooth=False):
+    # Fandom-change-end
```
Comment (adpaste) [inline thumbor/engines/pil.py]: Bad indentation.

**Comment by adpaste** [comment]
Not sure if it was intentional.

Our fork had uncommented config:
https://github.com/Wikia/thumbor/blob/master/thumbor/thumbor.conf#L134-L140

Whereas in this branch it's commented out:
https://github.com/Wikia/thumbor/blob/PLATFORM-7238-restore-our-changes/thumbor/thumbor.conf#L152-L157

File: thumbor/engines/pil.py [hunk 3]
```diff
@@ -525,12 +527,19 @@ def image_data_as_rgb(self, update_image=True):
...
         if "A" in self.image.mode and alpha:
```
Comment (mmeller-wikia) [inline thumbor/engines/pil.py]:
```suggestion
        # Fandom-change-end
        if "A" in self.image.mode and alpha:
```

**Comment by mmeller-wikia** [comment]
> Not sure if it was intentional.
> 
> Our fork had uncommented config: https://github.com/Wikia/thumbor/blob/master/thumbor/thumbor.conf#L134-L140
> 
> Whereas in this branch it's commented out: https://github.com/Wikia/thumbor/blob/PLATFORM-7238-restore-our-changes/thumbor/thumbor.conf#L152-L157

This configuration is overriden in smart-crop repository https://github.com/Wikia/smart-crop/blob/b3459c6c17d419f847cbec86a1858d6fc4cc2a78/thumbor/conf/thumbor.conf.tpl#L534
I left it commented out to reduce our fork :)

**Review by adpaste — APPROVED** [review]

**Review by adpaste — APPROVED** [review]

**Pull request merged**